src/node/mini_miner.cpp

```diff
@@ -0,0 +1,358 @@
+    // the target feerate:
+    //         target_feerate × ancestor_set_size - ancestor_set_fees
+    // By picking the maximum from the two, we ensure that a transaction meets
+    // both criteria.
+    for (const auto& [txid, outpoints] : m_requested_outpoints_by_txid) {
+        auto it = m_entries_by_txid.find(txid);
+        Assume(it != m_entries_by_txid.end());
+        if (it != m_entries_by_txid.end()) {
+            Assume(target_feerate.GetFee(it->second.GetSizeWithAncestors()) > std::min(it->second.GetModifiedFee(), it->second.GetModFeesWithAncestors()));
+            CAmount bump_fee_with_ancestors = target_feerate.GetFee(it->second.GetSizeWithAncestors()) - it->second.GetModFeesWithAncestors();
+            CAmount bump_fee_individual = target_feerate.GetFee(it->second.GetTxSize()) - it->second.GetModifiedFee();
+            const CAmount bump_fee{std::max(bump_fee_with_ancestors, bump_fee_individual)};
+            Assume(bump_fee >= 0);
+            for (const auto& outpoint : outpoints) {
+                m_bump_fees.emplace(outpoint, bump_fee);
+            }
+        }
+    }
+    return m_bump_fees;
+}
+
+std::optional<CAmount> MiniMiner::CalculateTotalBumpFees(const CFeeRate& target_feerate)
+{
+    if (!m_ready_to_calculate) return std::nullopt;
+    // Build a block template until the target feerate is hit.
+    BuildMockTemplate(target_feerate);
+
+    // All remaining ancestors that are not part of m_in_block must be bumped, but no other relatives
+    std::set<MockEntryMap::iterator, IteratorComparator> ancestors;
+    {
+        std::set<MockEntryMap::iterator, IteratorComparator> to_process;
+        for (const auto& [txid, outpoints] : m_requested_outpoints_by_txid) {
+            // Skip any ancestors that already have a miner score higher than the target feerate
+            // (already "made it" into the block)
+            if (m_in_block.count(txid)) continue;
+            auto iter = m_entries_by_txid.find(txid);
+            if (iter == m_entries_by_txid.end()) continue;
+            to_process.insert(iter);
+            ancestors.insert(iter);
+        }
+        while (!to_process.empty()) {
+            auto iter = to_process.begin();
+            assert(iter != to_process.end());
+            const CTransaction& tx = (*iter)->second.GetTx();
+            for (const auto& input : tx.vin) {
+                if (auto parent_it{m_entries_by_txid.find(input.prevout.hash)}; parent_it != m_entries_by_txid.end()) {
+                    to_process.insert(parent_it);
+                    ancestors.insert(parent_it);
+                }
+            }
+            to_process.erase(iter);
+        }
+    }
+    const auto ancestor_package_size = std::accumulate(ancestors.cbegin(), ancestors.cend(), 0,
+        [](int64_t sum, const auto it) {return sum + it->second.GetTxSize();});
+    const auto ancestor_package_fee = std::accumulate(ancestors.cbegin(), ancestors.cend(), 0,
```

Comment:
The init value should be `CAmount{0}`, not `0` which would be converted to an `int`, see [std::accumulate](https://en.cppreference.com/w/cpp/algorithm/accumulate) under "Common mistakes". This caused the unexpected fuzzer behavior I saw, because the `int` could overflow.
Also, for `ancestor_package_size`, the init value should be `int64_t{0}`;

Reply:
Uff, great catch, thanks!